src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+
+    Returns:
+        DNAnexus link or DXFile pointing to the uploaded file (depending on
+        the value of `return_handler`).
+
+    Raises:
+        ValueError: if compression_level not between 1 and 9
+        CalledProcessError: propogated from run_pipe if called command fails
+    """
+    ext = ".{}".format(method)
+    if remote_filename is None:
+        remote_filename = os.path.basename(filename)
+    if not remote_filename.endswith(ext):
+        remote_filename += ext
+
+    if not remote_folder:
+        remote_folder = "/"
+    elif not remote_folder.startswith("/"):
+        remote_folder = "/{}".format(remote_folder)
+
+    remote_path = "{}/{}".format(remote_folder, remote_filename)
+    if project:
+        remote_path = "{}:{}".format(project, remote_path)
+
+    # check that compression level is valid
+    if compression_level < 1 or compression_level > 9:
+        raise ValueError("Compression level must be between 1 and 9")
+
+    if not max_part_size or max_part_size <= 0:
+        max_part_size = _get_max_part_size(project)
+
+    if filename.endswith(ext):
+        cmd = [
+            "dx", "upload", "--brief", "--buffer-size", str(max_part_size),
+            "--path", remote_filename, filename
+        ]
+        file_id = proc.run_cmd(cmd).out
+    else:
+        exe = "bzip2" if method == "bz2" else "gzip"
+        zip_cmd = [exe, "-{0}".format(compression_level), "-c", filename]
+        upload_cmd = [
+            "dx", "upload", "--brief", "--buffer-size", str(max_part_size),
+            "--path", remote_path
+        ]
+        if wait_on_close:
+            upload_cmd.append("--wait")
+        upload_cmd.append("-")
+        file_id = proc.chain_cmds([zip_cmd, upload_cmd], shell=True).out
+
+    return _wrap_file_id(file_id, return_handler)
+
+
+def archive_and_upload_files(
+    filenames,
+    remote_prefix=None,
+    remote_folder="/",
+    compression_level=1,
+    chdir=None,
+    return_handler=False,
+    project=None,
+    wait_on_close=False,
+    method="gz",
+    max_part_size=None
+):
+    """
+    Archive and upload one or more files.
+
+    Shorthand for running 'tar', 'gzip', and 'dx upload' using subprocess on a
+    list of local files/directories.
+
+    Args:
+        filenames (str or list of str): = Local filenames or directories.
+        remote_prefix (str) = Name to give to output tar archive. Must be provided
+            unless `filenames` is of length 1, in which case the prefix will be
+            the same as that of the single filename.
+        remote_folder (str): Optional, remote folder to upload file to.
+        compression_level (int) = Level of compression between 1 and 9 to
+            compress tar to. Specify 1 for gzip --fast and 9 for gzip --best.
+            If not specified, --fast is assumed. If None, no compression is
+            performed, i.e. the output is a .tar file.
+        chdir (bool): If a path, change to this directory; if None and filenames is
```

Comment:
The usage of `chdir` as a boolean is confusing here. Probably would be better to have the actual path to cd to be specified, much like passing a dir to the `-C` flag. This also allows for other paths to be specified under the `tar -C ` command.